examples/vision/yolov8.py

```diff
@@ -0,0 +1,571 @@
+classes = tf.ragged.constant(classes)
+image_paths = tf.ragged.constant(image_paths)
+
+data = tf.data.Dataset.from_tensor_slices((image_paths, classes, bbox))
+
+"""
+Splitting data in training and validation data
+"""
+
+# Determine the number of validation samples
+num_val = int(len(xml_files) * SPLIT_RATIO)
+
+# Split the dataset into train and validation sets
+val_data = data.take(num_val)
+train_data = data.skip(num_val)
+
+"""
+Let's see about data loading and bounding box formatting to get things going. Bounding
+boxes in KerasCV have a predetermined format. To do this, you must bundle your bounding
+boxes into a dictionary that complies with the requirements listed below:
+
+```python
+bounding_boxes = {
+    # num_boxes may be a Ragged dimension
+    'boxes': Tensor(shape=[batch, num_boxes, 4]),
+    'classes': Tensor(shape=[batch, num_boxes])
+}
+```
+
+The dictionary has two keys, `'boxes'` and `'classes'`, each of which maps to a
+TensorFlow RaggedTensor or Tensor object. The 'boxes' Tensor has a shape of `[batch,
+num_boxes, 4]`, where batch is the number of images in the batch and num_boxes is the
+maximum number of bounding boxes in any image. The 4 represents the four values needed to
+define a bounding box:  xmin, ymin, xmax, ymax.
+
+The 'classes' Tensor has a shape of `[batch, num_boxes]`, where each element represents
+the class label for the corresponding bounding box in the 'boxes' Tensor. The num_boxes
+dimension may be ragged, which means that the number of boxes may vary across images in
+the batch.
+
+Final dict should be:
+```python
+{"images": images, "bounding_boxes": bounding_boxes}
+```
+"""
+
+
+def load_image(image_path):
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, channels=3)
+    return image
+
+
+def load_dataset(image_path, classes, bbox):
+    # Read Image
+    image = load_image(image_path)
+    bounding_boxes = {
+        "classes": tf.cast(classes, dtype=tf.float32),
+        "boxes": bbox,
+    }
+    return {"images": tf.cast(image, tf.float32), "bounding_boxes": bounding_boxes}
+
+
+"""
+Here we create a layer that resizes images to 640x640 pixels, while maintaining the
+original aspect ratio. The bounding boxes associated with the image are specified in the
+`xyxy` format. If necessary, the resized image will be padded with zeros to maintain the
+original aspect ratio.
+
+Bounding Box Formats supported by KerasCV:
+1.   CENTER_XYWH
+2.   XYWH
+3.   XYXY
+4.   REL_XYXY
+
+You can read more about KerasCV bounding box formats in
+[docs](https://keras.io/api/keras_cv/bounding_box/formats/).
+
+Also, you can convert one format to another using:
+
+```python
+boxes = keras_cv.bounding_box.convert_format(
+        bounding_box,
+        images=image,
+        source="xyxy",  # Original Format
+        target="xywh",  # Target Format (to which we want to convert)
+    )
+```
+
+Conversion you can do using KerasCV:
+
+
+1.   center_yxhw to xyxy
```

Comment:
I'm not sure if this is super helpful to include.

If we want to include it, we could also just list the formats and say that we can convert between any pair of them